locking/lockable.go

```diff
@@ -0,0 +1,209 @@
+package locking
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/config"
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/git-lfs/git-lfs/filepathfilter"
+	"github.com/git-lfs/git-lfs/git"
+	"github.com/git-lfs/git-lfs/tools"
+)
+
+// GetLockablePatterns returns a list of patterns in .gitattributes which are
+// marked as lockable
+func (c *Client) GetLockablePatterns() []string {
+	c.ensureLockablesLoaded()
+	return c.lockablePatterns
```

Comment:
This read isn't guarded by a mutex-lock. I think using a `sync.RWMutex` would be appropriate here.